--- initscripts-6.40/rc.d/init.d/halt.s390init	Wed Sep  5 05:18:34 2001
+++ initscripts-6.40/rc.d/init.d/halt	Fri Sep 14 17:36:28 2001
@@ -115,7 +115,6 @@
      ;;
 esac
 
-runcmd $"Syncing hardware clock to system time" /sbin/hwclock $CLOCKFLAGS
 
 # Turn off swap, then unmount file systems.
 SWAPS=`awk '! /^Filename/ { print $1 }' /proc/swaps`
--- initscripts-6.40/rc.d/rc.sysinit.s390init	Sun Sep  9 09:00:54 2001
+++ initscripts-6.40/rc.d/rc.sysinit	Fri Sep 14 17:39:17 2001
@@ -105,9 +105,8 @@
      ;;
 esac
 
-/sbin/hwclock $CLOCKFLAGS
-
-action $"Setting clock $CLOCKDEF: `date`" /bin/true
+# System date on S390 is always set correctly
+action $"System date: `date` " /bin/true
 
 if [ "$CONSOLETYPE" = "vt" ]; then
  # Load keymap
